1-js/02-first-steps/15-function-expressions/article.md

```diff
@@ -148,143 +148,143 @@ function ask(question, yes, no) {
 
-We can pass it between variables and run when we want.
+Możemy przekazywać ją pomiędzy zmiennymi i wywoływać kiedy chcemy.
 ```
 
 
-## Function Expression vs Function Declaration
+## Wyrażenie funkcyjne a deklaracja funkcji
 
-Let's formulate the key differences between Function Declarations and Expressions.
+Podsumujmy najważniejsze różnice między deklaracją funkcji a wyrażeniem funkcyjnym.
 
-First, the syntax: how to differentiate between them in the code.
+Po pierwsze, składnia - jak odróżnić je w kodzie.
 
-- *Function Declaration:* a function, declared as a separate statement, in the main code flow.
+- *Deklaracja funkcji:* funkcja, zadeklarowana jako osobne polecenie, w głównym ciągu kodu.
 
     ```js
-    // Function Declaration
+    // deklaracja funkcji
     function sum(a, b) {
       return a + b;
     }
     ```
-- *Function Expression:* a function, created inside an expression or inside another syntax construct. Here, the function is created at the right side of the "assignment expression" `=`:
+- *Wyrażenie funkcyjne:* funkcja, utworzona wewnątrz wyrażenia lub innej konstrukcji składniowej. Poniżej, funkcja jest tworzona po prawej stronie "wyrażenia przypisania" `=`:
 
     ```js
-    // Function Expression
+    // wyrażenie funkcyjne
     let sum = function(a, b) {
       return a + b;
     };
     ```
 
-The more subtle difference is *when* a function is created by the JavaScript engine.
+Bardziej subtelną różnicą jest różnica w *momencie tworzenia* funkcji przez silnik JavaScriptu.
 
-**A Function Expression is created when the execution reaches it and is usable only from that moment.**
+**Wyrażenie funkcyjne jest tworzone w momencie jego napotkania podczas wykonywania skryptu. Można używać go tylko od tej chwili.**
 
-Once the execution flow passes to the right side of the assignment `let sum = function…` -- here we go, the function is created and can be used (assigned, called, etc. ) from now on.
+Kiedy wykonanie skryptu dojdzie do prawej strony przypisania `let sum = function…`, funkcja jest tworzona i od tego momentu może być używana (przypisana, wywołana, itp.).
 
-Function Declarations are different.
+Deklaracje funkcji działają inaczej.
 
-**A Function Declaration can be called earlier than it is defined.**
+**Funkcja deklarowana może być wywołana zanim zostanie zdefiniowana.**
 
-For example, a global Function Declaration is visible in the whole script, no matter where it is.
+Globalna deklaracja funkcji jest widoczna **w całym** skrypcie, niezależnie od tego, w którym miejscu jest zadeklarowana.
 
-That's due to internal algorithms. When JavaScript prepares to run the script, it first looks for global Function Declarations in it and creates the functions. We can think of it as an "initialization stage".
+Wynika to z algorytmu wykonywania skryptu. Przygotowując skrypt do wykonania, JavaScript najpierw wyszukuje w nim globalnych deklaracji funkcji, a następnie je tworzy - można nazwać to etapem "inicjalizacji".
```

Comment:
Replace `Wynika to z algorytmu wykonywania skryptu.` by `Wynika to z wewnętrznego algorytmu wykonywania skryptu.`